integration/apiserver.go

```diff
@@ -65,7 +60,7 @@ type APIServer struct {
 
 // Start starts the apiserver, waits for it to come up, and returns an error,
 // if occurred.
-func (s *APIServer) Start() error {
+func (s *APIServer) Start(etcdConnectionConfig RemoteConnectionConfig) error {
```

Comment:
Each component might take a different type of config, or provide a different set of configuration options. That's fine for now, but I'm afraid that this will be too generic very quickly.

Reply:
We **think** we might come a long way with a shared config. But yeah, when we get proven wrong we need to have multiple config types.